homeassistant/components/weatherflow/sensor.py

```diff
@@ -0,0 +1,401 @@
+        """Post initialisation processing."""
+        self.native_unit_of_measurement = TEMP_CELSIUS
+        self.device_class = SensorDeviceClass.TEMPERATURE
+        self.state_class = SensorStateClass.MEASUREMENT
+        self.decimals = 1
+
+
+@dataclass
+class WeatherFlowWindSensorEntityDescription(WeatherFlowSensorEntityDescription):
+    """Describes a WeatherFlow wind sensor entity description."""
+
+    def __post_init__(self) -> None:
+        """Post initialisation processing."""
+        self.icon = "mdi:weather-windy"
+        self.native_unit_of_measurement = SPEED_KILOMETERS_PER_HOUR
+        self.state_class = SensorStateClass.MEASUREMENT
+        self.conversion_fn = lambda attr: attr.to(SPEED_MILES_PER_HOUR)
+        self.decimals = 2
+        self.value_fn = lambda attr: attr.to(QUANTITY_KILOMETERS_PER_HOUR)
+
+
+SENSORS: tuple[WeatherFlowSensorEntityDescription, ...] = (
+    WeatherFlowTemperatureSensorEntityDescription(
+        key="air_temperature",
+        name="Temperature",
+        has_entity_name=True,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="air_density",
+        name="Air density",
+        has_entity_name=True,
+        native_unit_of_measurement=CONCENTRATION_KILOGRAMS_PER_CUBIC_METER,
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        state_class=SensorStateClass.MEASUREMENT,
+        conversion_fn=lambda attr: attr.to(CONCENTRATION_POUNDS_PER_CUBIC_FOOT),
+        decimals=5,
+    ),
+    WeatherFlowTemperatureSensorEntityDescription(
+        key="dew_point_temperature",
+        name="Dew point",
+        has_entity_name=True,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="battery",
+        name="Battery voltage",
+        has_entity_name=True,
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WeatherFlowTemperatureSensorEntityDescription(
+        key="feels_like_temperature",
+        name="Feels like",
+        has_entity_name=True,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="illuminance",
+        name="Illuminance",
+        has_entity_name=True,
+        native_unit_of_measurement=LIGHT_LUX,
+        device_class=SensorDeviceClass.ILLUMINANCE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="lightning_strike_average_distance",
+        name="Lightning average distance",
+        has_entity_name=True,
+        icon="mdi:lightning-bolt",
+        native_unit_of_measurement=LENGTH_KILOMETERS,
+        conversion_fn=lambda attr: attr.to(LENGTH_MILES),
+        decimals=2,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="lightning_strike_count",
+        name="Lightning count",
+        has_entity_name=True,
+        icon="mdi:lightning-bolt",
```

Comment:
It would be nice if the following line could be added here to allow a running count of lightning strikes similar to rainfall amount.

```
        state_class=SensorStateClass.TOTAL,
```

Reply:
@natekspencer - thoughts?

Reply:
Yeah, we should add it. It was probably just an oversight from me on not including it in the first place.